### DISCUSSION TITLE
Popup on click

### DISCUSSION THREAD

**Discussion by bramborman** [pcolby/nfc-quick-settings#49] [Ideas]
The stock Android UI, that is, the one on Google Pixel and some other devices has a popup that opens when clicking on some of the Quick Settings tiles, such as Bluetooth or Internet. The popup allows one to turn Bluetooth/Wi-Fi/Data On/Off, but also gives a quick access to connect to paired devices/Wi-Fi networks.
It would be nice to have such a feature for this app, too. It wouldn't make any sense if the app didn't include anything besides toggle functionality, but together with other features such as automatic turn off (#48) it would be really helpful.
As for the implementation part, it's likely not possible to do exactly what the system UI does - the Quick Settings tile animates away and becomes the popup, but with a "display on top of other apps" permission the app could achieve something similar, just without the animation.

For a reference, [here's some article](https://9to5google.com/2022/02/24/how-to-use-android-12s-new-internet-quick-settings-tile/) I found with screenshots from Android 12 depicting the Internet popup design. It doesn't differ much in Android 14.

**Comment by corobin** [comment]
Bluetooth has moved to that format because you can also choose bluetooth devices. NFC doesn't really have any similar type of interaction that you want to be available at the surface, the dominant purpose of this app really is just the toggle. So i don't think the pop up UI would be very suitable. stuff like configuring auto-shut-off should be left to a UI somewhere else, not immediately in the pop-up panel, since putting it in the pop-up would come at the cost of quickly being able to toggle it, which detracts from the reason the app exists in the first place.
